File: backend/src/main/java/com/voyago/service/DestinationService.java
package com.voyago.service;

import com.voyago.model.Destination;
import com.voyago.repository.DestinationRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class DestinationService {

    private final DestinationRepository destinationRepository;

    public List<Destination> getAll() {
        return destinationRepository.findAll();
    }

    public Destination getById(String id) {
        return destinationRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Destination not found: " + id));
    }

    public List<Destination> getTrending() {
        return destinationRepository.findByIsTrendingTrue();
    }

    public List<Destination> search(String name) {
        return destinationRepository.findByNameContainingIgnoreCase(name);
    }

    public Destination create(Destination destination) {
        return destinationRepository.save(destination);
    }

    public Destination update(String id, Destination updated) {
        Destination existing = getById(id);
        updated.setId(existing.getId());
        return destinationRepository.save(updated);
    }

    public void delete(String id) {
        destinationRepository.deleteById(id);
    }
}

File: src/services/destinationService.ts
import apiClient from './api';
import { Destination } from '../types';

export const destinationService = {
  async getAll(): Promise<Destination[]> {
    const { data } = await apiClient.get<Destination[]>('/destinations');
    return data;
  },

  async getTrending(): Promise<Destination[]> {
    const { data } = await apiClient.get<Destination[]>('/destinations/trending');
    return data;
  },

  async search(name: string): Promise<Destination[]> {
    const { data } = await apiClient.get<Destination[]>('/destinations/search', { params: { name } });
    return data;
  },

  async getById(id: string): Promise<Destination> {
    const { data } = await apiClient.get<Destination>(`/destinations/${id}`);
    return data;
  },

  async create(destination: Omit<Destination, 'id'>): Promise<Destination> {
    const { data } = await apiClient.post<Destination>('/destinations', destination);
    return data;
  },

  async update(id: string, destination: Partial<Destination>): Promise<Destination> {
    const { data } = await apiClient.put<Destination>(`/destinations/${id}`, destination);
    return data;
  },

  async delete(id: string): Promise<void> {
    await apiClient.delete(`/destinations/${id}`);
  },
};

export default destinationService;
